### DISCUSSION TITLE
API rule that ensures uniqueness of field based on another field

### DISCUSSION THREAD

**Discussion by zakkor** [pocketbase/pocketbase#2314] [Q&A] [answered]
Concrete example:

We have the `users` collection, which contains a `company` (relation) field. Users have a `name`, and the name needs to be unique *per company*, not across the entire collection (which could be achieved using the Unique modifier on the column)

Right now I'm doing a separate request
```
.getList(1, 1, { filter: `company = "${user.company}" && name = "${name}"` })
```
and showing an error if there is any result.

Would it be possible to do this using an API rule directly?

**Comment by ganigeorgiev** [comment, marked as answer]
You can create a unique index with both `name` and `company` fields (_available with PocketBase v0.14.0+_), aka. the result statement should look something like:

```sql
CREATE UNIQUE INDEX `idx_RTC5POj` ON `users` (
  `name`,
  `company`
)
```

**Reply by zakkor** [reply]
Oh, I was using Pocketbase 0.13, so I didn't see that option. I've upgraded now & the suggested index works perfectly, thanks!
